### PR TITLE
Add doc for per-column compression codecs

### PR DESCRIPTION
**Pull request by alex-krash**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Other

Short description (up to few sentences):
Documentation for new feature (per-column compression codecs)
...

Detailed description (optional):

...

### CODE REVIEW

File: docs/en/operations/table_engines/custom_compression_codec.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+# Column compression codecs
```
Comment (blinkov) [inline docs/en/operations/table_engines/custom_compression_codec.md]: Title case in English headers is used: Column Compression Codecs

File: docs/en/operations/table_engines/custom_compression_codec.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    ts DateTime CODEC(LZ4HC),
+    float_value Float32 CODEC(NONE),
+    double_value Float64 CODEC(LZ4HC(2))
+)
+ENGINE = MergeTree
+PARTITION BY tuple()
+ORDER BY dt
+```
+
+Codecs can be combined in a pipeline. Example below shows an optimization approach for storing timeseries metrics.
```
Comment (blinkov) [inline docs/en/operations/table_engines/custom_compression_codec.md]: Needs clarification that default compression codec is not included in pipeline.

File: docs/en/operations/table_engines/custom_compression_codec.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+
+# Column compression codecs
+
+Besides default data compression, defined in [server settings](../server_settings/settings.md#compression), per-column specification is also available.
+
+Supported compression algorithms:
+
+- `NONE` - no compression for data applied
+- `LZ4`
+- `LZ4HC(level)` - (level) - LZ4 compression algorithm with defined level.
```
Comment (alesapin) [inline docs/en/operations/table_engines/custom_compression_codec.md]: Need to add information about default levels: LZ4HC -- 9, ZSTD -- 1, Delta -- sizeof(type) if type size is 1, 2, 4, 8 and 1 in other cases.

File: docs/en/operations/table_engines/custom_compression_codec.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+Possible `level` range: [1,12]. Greater values stands for better compression and higher CPU usage. Recommended value range: [4,9].
+- `ZSTD(level)` - ZSTD compression algorithm with defined `level`. Possible `level` value range: [1,22].
+Greater values stands for better compression and higher CPU usage.
+- `Delta(delta_bytes)` - compression approach when raw values are replace with difference with two neighbour values. Up to `delta_bytes` are used for storing delta value.
```
Comment (alesapin) [inline docs/en/operations/table_engines/custom_compression_codec.md]: with difference **of** two neighbour values?

File: docs/en/operations/table_engines/custom_compression_codec.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+# Column compression codecs
+
+Besides default data compression, defined in [server settings](../server_settings/settings.md#compression), per-column specification is also available.
+
+Supported compression algorithms:
+
+- `NONE` - no compression for data applied
+- `LZ4`
+- `LZ4HC(level)` - (level) - LZ4 compression algorithm with defined level.
+Possible `level` range: [1,12]. Greater values stands for better compression and higher CPU usage. Recommended value range: [4,9].
```
Comment (alesapin) [inline docs/en/operations/table_engines/custom_compression_codec.md]: https://github.com/lz4/lz4/blob/c10863b98e1503af90616ae99725ecd120265dfb/lib/lz4hc.h#L46-L50 [3-12], in other cases 9.

File: docs/en/operations/table_engines/custom_compression_codec.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+Possible `delta_bytes` values: 1, 2, 4, 8;
+
+Syntax example:
+```
+CREATE TABLE codec_example
+(
+    dt Date CODEC(ZSTD), /* используется уровень сжатия по-умолчанию */
+    ts DateTime CODEC(LZ4HC),
+    float_value Float32 CODEC(NONE),
+    double_value Float64 CODEC(LZ4HC(2))
```
Comment (alesapin) [inline docs/en/operations/table_engines/custom_compression_codec.md]: Bad example, actually will be LZ4HC(9)

File: docs/en/operations/table_engines/custom_compression_codec.md (hunk 3 above)
Comment (alesapin) [reply docs/en/operations/table_engines/custom_compression_codec.md]: >  - `LZ4HC(level)` - >>**(level)**<< - LZ4 compression algorithm with defined level. 

typo?

**Review by alesapin — APPROVED** [review]

**Pull request merged**